spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicConfigurationSupport.java

```diff
@@ -0,0 +1,485 @@
+import org.springframework.kafka.retrytopic.DestinationTopic;
+import org.springframework.kafka.retrytopic.DestinationTopicProcessor;
+import org.springframework.kafka.retrytopic.DestinationTopicResolver;
+import org.springframework.kafka.retrytopic.ListenerContainerFactoryConfigurer;
+import org.springframework.kafka.retrytopic.ListenerContainerFactoryResolver;
+import org.springframework.kafka.retrytopic.RetryTopicBeanNames;
+import org.springframework.kafka.retrytopic.RetryTopicBootstrapper;
+import org.springframework.kafka.retrytopic.RetryTopicConfigurer;
+import org.springframework.kafka.retrytopic.RetryTopicInternalBeanNames;
+import org.springframework.kafka.retrytopic.RetryTopicNamesProviderFactory;
+import org.springframework.kafka.support.JavaUtils;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.util.Assert;
+import org.springframework.util.backoff.BackOff;
+import org.springframework.util.backoff.FixedBackOff;
+
+/**
+ * This is the main class providing the configuration behind the non-blocking,
+ * topic-based delayed retries feature. It is typically imported by adding
+ * {@link EnableRetryTopic @EnableRetryTopic} to an application
+ * {@link Configuration @Configuration} class. An alternative more advanced option
+ * is to extend directly from this class and override methods as necessary, remembering
+ * to add {@link Configuration @Configuration} to the subclass and {@link Bean @Bean}
+ * to overridden {@link Bean @Bean} methods. For more details see the javadoc of
+ * {@link EnableRetryTopic @EnableRetryTopic}.
+ *
+ * @author Tomaz Fernandes
+ * @since 2.9
+*/
+public class RetryTopicConfigurationSupport {
+
+	private final RetryTopicComponentFactory componentFactory = createComponentFactory();
+
+	private static final String BACK_OFF_MANAGER_THREAD_EXECUTOR_BEAN_NAME = "backOffManagerThreadExecutor";
+
+	/**
+	 * Return a global {@link RetryTopicConfigurer} for configuring retry topics
+	 * for {@link KafkaListenerEndpoint} instances with a corresponding
+	 * {@link org.springframework.kafka.retrytopic.RetryTopicConfiguration}.
+	 * To configure it, consider overriding the {@link #configureRetryTopicConfigurer()}.
+	 * @param kafkaConsumerBackoffManager the global {@link KafkaConsumerBackoffManager}.
+	 * @param destinationTopicResolver the global {@link DestinationTopicResolver}.
+	 * @param beanFactory the {@link BeanFactory}.
+	 * @return the instance.
+	 * @see KafkaListenerAnnotationBeanPostProcessor
+	 */
+	@Bean(name = RetryTopicBeanNames.RETRY_TOPIC_CONFIGURER_BEAN_NAME)
+	public RetryTopicConfigurer retryTopicConfigurer(@Qualifier(KafkaListenerConfigUtils.KAFKA_CONSUMER_BACK_OFF_MANAGER_BEAN_NAME)
+																KafkaConsumerBackoffManager kafkaConsumerBackoffManager,
+													@Qualifier(RetryTopicBeanNames.DESTINATION_TOPIC_RESOLVER_BEAN_NAME)
+															DestinationTopicResolver destinationTopicResolver,
+													BeanFactory beanFactory) {
```

Comment:
We don't use this indentation style; the additional parameters (and the first in this case) should start at 2 tabs past the `public`.